tests/cli/main.cpp

```diff
@@ -197,252 +197,218 @@ class client_connection
 };
 
 ///////////////////////////////
-// Tests
+// Cli Wallet Fixture
 ///////////////////////////////
 
-////////////////
-// Start a server and connect using the same calls as the CLI
-////////////////
-BOOST_AUTO_TEST_CASE( cli_connect )
+struct cli_fixture
 {
-   using namespace graphene::chain;
-   using namespace graphene::app;
-   std::shared_ptr<graphene::app::application> app1;
-   try {
-      fc::temp_directory app_dir ( graphene::utilities::temp_directory_path() );
+   int server_port_number {0};
+   fc::temp_directory app_dir = graphene::utilities::temp_directory_path();
+   std::shared_ptr<graphene::app::application> app1 = start_application(app_dir, server_port_number);
+   client_connection* con = new client_connection(app1, app_dir, server_port_number);
```

Comment:
Initializing these variables here is a bad idea. They're heavy. Better move them to a constructor. Then some member variables can be changed to local variable.

By the way, why need to use a pointer?

Reply:
1. fixed indentation from 4 white spaces to 3
2. initializing variables in constructor has different compile errors, need to fight all them, trying to understand why they happened, still fighting ...
3. I've used pointer because without pointer I had compile errors, still trying to understand why and still fighting with them ...

Reply:
unknown location(0): fatal error: in "cli_connect": memory access violation at address: 0x00000064: no mapping at fault address
/home/valera/Projects/Blockchain/Graphene/BitShares/bitshares-core/tests/cli/main.cpp(260): last checkpoint: "cli_connect" fixture ctor

*** 1 failure is detected in the test module "Test Application"

such error I see when trying to call next method:

`app_dir = graphene::utilities::temp_directory_path();`

It means next:

`BOOST_FIXTURE_TEST_CASE( cli_connect, cli_fixture )` - this Boost fixture cannot call `cli_fixture` constructor with `graphene::utilities::temp_directory_path()` method inside, maybe
I'm doing something wrong ? Still trying to understand the reason and to code as we desire !

Reply:
fixed by direct initialization instead of using assignment operators inside constructor body